apps/speed.c

```diff
@@ -933,11 +1459,28 @@ int speed_main(int argc, char **argv)
         goto end;
     }
 
+    if (async_jobs > 0) {
+        if (!ASYNC_init(1, 0, 0)) {
+            BIO_printf(bio_err, "Error creating the ASYNC job pool\n");
+            goto end;
+        }
+    }
+
+    array_loopargs = app_malloc((!async_jobs ? 1 : async_jobs) * sizeof(loopargs*), "array of loopargs*");
+    for (i = 0; i < (!async_jobs ? 1 : async_jobs); i++) {
+        array_loopargs[i] = app_malloc((sizeof(loopargs)), "an individual looparg structure");
+        memset(array_loopargs[i], 0, sizeof(loopargs));
+    }
```

Comment:
Why the indirection of an array of loopargs* each pointing at a loopargs? Why not just create an array of loopargs straight off?